### DISCUSSION TITLE
Not all linux distributions have /bin/{shell} (e.g. NixOS)

### DISCUSSION THREAD

**Discussion by sweenu** [Taitava/obsidian-shellcommands#296] [Ideas - done & released]
Several Linux distributions do not follow the [FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard). 
Here's how it looks like for me on NixOS:
```console
$ echo $SHELL
/run/current-system/sw/bin/fish
$ readlink /run/current-system/sw/bin/fish
/nix/store/q8f2zk07qi7hmvgppdsfa1pwiihyrnji-fish-3.5.1/bin/fish
$ ls /bin/bash
ls: cannot access '/bin/bash': No such file or directory
$ which bash
/run/current-system/sw/bin/bash
$ readlink (which bash)
/nix/store/bap4d0lpcrhpwmpb8ayjcgkmvfj62lnq-bash-interactive-5.1-p16/bin/bash
```
All executables on NixOS are in my nix store (/nix/store) and are symbolically linked to some other places (here for example in /run/current-system/sw, which is in my $PATH).

So the only way for me to use that plugin is to manually `ln -s /run/current-system/sw/bin/bash /bin/bash`.  And choose the bash option in the plugin's settings. It's not ideal, it's more of a hack that I truly wish I could avoid doing just for one obsidian plugin.

There is a real need for https://github.com/Taitava/obsidian-shellcommands/discussions/108 or at least adding an option for `/bin/sh` because this will always be present in NixOS and I figure also in other such distros.

Love the plugin otherwise, very good job!

**Comment by Taitava** [comment]
Hi, thank you for your post and for your feedback! 🙂 

You are right, #108 is definitely needed. I can't give any timetable for it, but I think it could be the next big thing to aim for. I've been thinking about it quite a lot recently.

**Reply by Taitava** [reply]
I moved this discussion from _Q&A_ to _Ideas_.
